Source/WebKit/UIProcess/Extensions/WebExtensionContext.h

```diff
@@ -963,6 +967,8 @@ class WebExtensionContext : public API::ObjectImpl<API::Object::Type::WebExtensi
     RetainPtr<_WKWebExtensionStorageSQLiteStore> m_localStorageStore;
     RetainPtr<_WKWebExtensionStorageSQLiteStore> m_sessionStorageStore;
     RetainPtr<_WKWebExtensionStorageSQLiteStore> m_syncStorageStore;
+
+    inline static bool s_hasOpenPopup { false };
```

Comment:
As we talked about, this shouldn't be a static. It would be best to put this on `WebExtensionController` so the state isn't global, while still preventing multiple extensions in the same controller (profile in Safari) from opening a popup while another is already showing.